impeller/renderer/render_pass.h

```diff
@@ -60,6 +63,81 @@ class RenderPass {
+  void SetPipeline(
+      const std::shared_ptr<Pipeline<PipelineDescriptor>>& pipeline);
+
+  //----------------------------------------------------------------------------
+  /// The debugging label to use for the command.
+  void SetCommandLabel(const std::string& label);
+
+  //----------------------------------------------------------------------------
+  /// The reference value to use in stenciling operations. Stencil configuration
+  /// is part of pipeline setup and can be read from the pipelines descriptor.
+  ///
+  /// @see         `Pipeline`
+  /// @see         `PipelineDescriptor`
+  ///
+  void SetStencilReference(uint32_t value);
+
+  void SetBaseVertex(uint64_t value);
+
+  //----------------------------------------------------------------------------
+  /// The viewport coordinates that the rasterizer linearly maps normalized
+  /// device coordinates to.
+  /// If unset, the viewport is the size of the render target with a zero
+  /// origin, znear=0, and zfar=1.
+  ///
+  void SetViewport(Viewport viewport);
+
+  //----------------------------------------------------------------------------
+  /// The scissor rect to use for clipping writes to the render target. The
+  /// scissor rect must lie entirely within the render target.
+  /// If unset, no scissor is applied.
+  ///
+  void SetScissor(IRect scissor);
+
+  //----------------------------------------------------------------------------
+  /// The number of instances of the given set of vertices to render. Not all
+  /// backends support rendering more than one instance at a time.
+  ///
+  /// @warning      Setting this to more than one will limit the availability of
+  ///               backends to use with this command.
+  ///
+  void SetInstanceCount(size_t count);
+
+  //----------------------------------------------------------------------------
+  /// @brief      Specify the vertex and index buffer to use for this command.
+  ///
+  /// @param[in]  buffer  The vertex and index buffer definition. If possible,
+  ///             this value should be moved and not copied.
+  ///
+  /// @return     returns if the binding was updated.
+  ///
+  bool SetVertexBuffer(VertexBuffer buffer);
+
+  /// Record the currently pending command.
+  bool Dispatch();
+
+  // |ResourceBinder|
+  bool BindResource(ShaderStage stage,
+                    const ShaderUniformSlot& slot,
+                    const ShaderMetadata& metadata,
+                    BufferView view) override;
+
+  bool BindResource(ShaderStage stage,
+                    const ShaderUniformSlot& slot,
+                    const std::shared_ptr<const ShaderMetadata>& metadata,
+                    BufferView view);
+
+  // |ResourceBinder|
+  bool BindResource(ShaderStage stage,
+                    const SampledImageSlot& slot,
+                    const ShaderMetadata& metadata,
+                    std::shared_ptr<const Texture> texture,
+                    std::shared_ptr<const Sampler> sampler) override;
```

Comment:
API to encode commands on the `RenderPass` directly LGTM. (And the generated reflection headers will also change to operate on the `RenderPass` instead of `Command`, of course.)